library/network_connections.py

```diff
@@ -1026,8 +1036,19 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_IP_CONFIG_ROUTE_METRIC, ip["route_metric6"]
                 )
             for d in ip["dns"]:
-                if d["family"] == socket.AF_INET6:
-                    s_ip6.add_dns(d["address"])
+                if d["family"] != socket.AF_INET6:
+                    continue
+                if ipv6_is_disabled:
+                    raise MyError(
+                        "when IPv6 is disabled, DNS setting for IPv6 nameservers is not allowed."
```

Comment:
```suggestion
                        "IPv6 needs to be enabled to support IPv6 nameservers."
```

Reply:
I changed into "IPv6 needs to be enabled to support IPv6 nameservers."